File: examples/python/01_autenticacion_y_sesion/auth_session.py
import requests
import base64
import json

API_URL = 'http://localhost:8080/api/v1'
API_KEY = 'B5D31933-C996-476C-B116-EF212A41479A'
API_ID = '1093'
USER = '001'
PASSWORD = '12345'

def iniciar_sesion():
    endpoint = API_URL + '/auth/login'

    headers = {
        'x-api-key': API_KEY,
        'x-api-id': API_ID
    }

    body = {
        "username": USER,
        "password": PASSWORD,
        "device_id":"Python",
    }

    try:
        response = requests.post(endpoint, headers=headers, json=body)
        response.raise_for_status()
        jsonResonse = response.json()

        if jsonResonse.get('success') and 'data' in jsonResonse:
            return jsonResonse['data']['access_token']
        else:
            return None
    except requests.exceptions.RequestException as e:
        return e

if __name__ == "__main__":
    token = iniciar_sesion()
    print(token)

File: examples/javascript/01_autenticacion_fetch_axios/auth_session.js
const API_URL = 'http://localhost:8080/api/v1'; 
const API_KEY = 'B5D31933-C996-476C-B116-EF212A41479A';
const API_ID = '1093';

const USER = '001';
const PASSWORD = '12345';
const DEVICE_ID = 'NODEJS_CLIENT_01';

async function iniciarSesion() {
    const endpoint = `${API_URL}/auth/login`;

    const headers = {
        'Content-Type': 'application/json',
        'x-api-key': API_KEY,
        'x-api-id': API_ID
    };

    const body = JSON.stringify({
        username: USER,
        password: PASSWORD,
        device_id: DEVICE_ID
    });

    try {
        console.log(`Intentando iniciar sesión en: ${endpoint}`);
        
        const response = await fetch(endpoint, {
            method: "POST",
            headers: headers,
            body: body
        });

        if (!response.ok) {
            const errorTxt = await response.text();
            throw new Error(`Error del servidor (${response.status}): ${errorTxt}`);
        }

        const resultado = await response.json();

        if (resultado.success && resultado.data && resultado.data.access_token) {
            console.log('Sesión establecida exitosamente.');
            return resultado.data.access_token;
            
        } else {
            console.error('Fallo en el login:', resultado.message || 'Respuesta inválida');
            return null;
        }

    } catch (error) {
        console.error('Error de comunicación:', error.message);
        return null;
    }
}

if (require.main === module) {
    iniciarSesion().then(token => {
        if (token) {
            console.log(`Token recibido: ${token}`);
        }
    });
}

module.exports = { iniciarSesion, API_URL };